tools/dump_kubernetes.sh

```diff
@@ -275,20 +315,59 @@ dump_resources() {
   mkdir -p "${OUT_DIR}"
   kubectl cluster-info dump > "${OUT_DIR}/cluster-info.dump.txt"
   kubectl describe pods -n istio-system > "${OUT_DIR}/istio-system-pods.txt"
-  kubectl get events --all-namespaces -o wide > "${OUT_DIR}/events.txt"
+  if [ -z "${namespaces}" ]; then
+    kubectl get events --all-namespaces -o wide > "${OUT_DIR}/events.txt"
+  else
+    for namespace in ${namespaces}; do
+      NAMESPACE_EVENT_FILE="${OUT_DIR}/${namespace}-events.txt"
+      kubectl get events -n "${namespace}" -o wide > "${NAMESPACE_EVENT_FILE}"
+    done
+  fi
+}
+
+dump_istio_status() {
+  istioctl version > "${OUT_DIR}/istioctl-version.txt"
+  istioctl proxy-status > "${OUT_DIR}/istioctl-proxy-status.txt"
+  kubectl get configmap -n istio-system > "${OUT_DIR}/istio-system-configmap.txt"
+  kubectl get configmap -n istio-system > "${OUT_DIR}/istio-system-configmap.txt"
+  kubectl get configmap istio -n istio-system && kubectl get configmap istio-sidecar-injector -n istio-system > "${OUT_DIR}/istio-system-configmaps.txt"
+}
+
+dump_control_plane() {
+  # dump_control_plane Should be called for only one namespace (and that would
+  # be "istio-system").
+  # This ns variable is kept to use for other namespaces if needed later
+  local ns="${namespaces}"
+  local describe_dir="${OUT_DIR}/describe"
+  mkdir -p "${describe_dir}"
+  kubectl -n "${ns}" get pods -o yaml > "${OUT_DIR}/${ns}-pods.yaml"
+  api_resources=$(kubectl api-resources -n "${ns}" -o name)
```

Comment:
I think this is too much, probably don't want secrets, etc and makes it slow. Maybe we should just do `{deployment,service,configmap}`? If we do configmap it makes the above configmap ones not needed as well.

Reply:
deployment,services*,configmaps* ? That's it?